### PR TITLE
DRIVERS-2328 download legacy shell from 5.0 package

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary
- If legacy "mongo" shell is not included, download from 5.0 packages.

These changes are tested in the [C driver here](https://spruce.mongodb.com/version/6282a68c3066155b78dbb43d/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) and the [Go driver here](https://spruce.mongodb.com/version/6280e17c2a60ed3484cc8bb6/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

# Background & Motivation

[SERVER-64352](https://jira.mongodb.org/browse/SERVER-64352) removes the legacy shell from server packages. Downloading from 5.0 is a temporary solution.

### CODE REVIEW

File: .evergreen/download-mongodb.sh [hunk 1]
```diff
@@ -504,3 +505,31 @@ download_and_extract ()
...
+{
+   MONGODB_DOWNLOAD_URL=$1
+   EXTRACT=$2
+
+   download_and_extract_package "$MONGODB_DOWNLOAD_URL" "$EXTRACT"
+
+   if [ ! -e $DRIVERS_TOOLS/mongodb/bin/mongo ]; then
+      # The legacy mongo shell is not included in server downloads of 6.0.0-rc6 or later. Refer: SERVER-64352.
+      # Some test scripts use the mongo shell for setup.
+      # Download 6.0 package to get the legacy mongo shell as a workaround until DRIVERS-2328 is addressed.
```
Comment (ShaneHarvey) [inline .evergreen/download-mongodb.sh:521]: If 6.0 does not include the legacy shell don't we need to download it from 5.0?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: .evergreen/download-mongodb.sh [hunk 2]
```diff
@@ -504,3 +505,31 @@ download_and_extract ()
...
+# download_and_extract downloads a requested MongoDB server package.
+# If the legacy shell is not included in the download, the legacy shell is also downloaded from the 6.0 package.
+download_and_extract ()
+{
+   MONGODB_DOWNLOAD_URL=$1
+   EXTRACT=$2
+
+   download_and_extract_package "$MONGODB_DOWNLOAD_URL" "$EXTRACT"
+
+   if [ ! -e $DRIVERS_TOOLS/mongodb/bin/mongo ]; then
```
Comment (vector-of-bool) [inline .evergreen/download-mongodb.sh:518]: Does this need to work on Windows, wherein the file will have a `.exe` suffix?

Comment (kevinAlbs) [reply .evergreen/download-mongodb.sh]: Great question. Yes. This has been updated. Curiously the `cp` command did not error on the [Windows test run](https://evergreen.mongodb.com/lobster/evergreen/task/mongo_c_driver_windows_2017_test_latest_server_auth_sspi_winssl_patch_1841a0eb0d344abf7bd84413807b1e15e3c92496_6280e09830661571add434b5_22_05_15_11_14_34/0/task#bookmarks=0%2C6252&shareLine=151). Updated to also check for `.exe`.

File: .evergreen/download-mongodb.sh (hunk 1 above)
Comment (kevinAlbs) [reply .evergreen/download-mongodb.sh]: Oof, good point. Updated to use 5.0.

**Review by vector-of-bool — APPROVED** [review]
LGTM

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**